### PR TITLE
New README

### LINKED ISSUE #990: Update README

### PR DESCRIPTION
**Pull request by rasbt**
Here's a draft for the new README. It's based on some great ideas from Adrian's initial draft, like showing only the 3 most recent models. I updated the docs a bit to keep the README more minimal. 

I think the challenge of the old README was that it was too long to read, and most people probably found it overwhelming. There was also lots of duplicated information between the README and the tutorials, and it might be better to not try to show everything in the README but rather show one or two examples and then reference the tutorials.

What are your thoughts?

Fixes #990

### CODE REVIEW

File: tutorials/finetune.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Understanding finetuning -- conceptual tutorials
+
+Looking for conceptual tutorials and explanations? We have some additional articles below:
+
+- [Understanding Parameter-Efficient Finetuning of Large Language Models: From Prefix Tuning to LLaMA-Adapters](https://lightning.ai/pages/community/article/understanding-llama-adapters/)
+
+- [Parameter-Efficient LLM Finetuning With Low-Rank Adaptation (LoRA)](https://lightning.ai/pages/community/tutorial/lora-llm/)
+
+
+&nbsp;
```
Comment (carmocca) [inline tutorials/finetune.md:50]:
```suggestion
```

File: tutorials/models.md
Comment (carmocca) [inline tutorials/models.md:1]: Is this file intended? You also added this info to `download_model_weights.md`

File: README.md [hunk 2]
```diff
@@ -14,183 +14,315 @@
...
-This repository follows the main principle of **openness through clarity**.
+The following [Lightning Studio](https://lightning.ai/lightning-ai/studios) templates provide LitGPT tutorials and projects in reproducible environments with multi-GPU and multi-node support:
 
-**LitGPT** is:
 
-- **Simple:** Single-file implementation without boilerplate.
-- **Correct:** Numerically equivalent to the original model.
-- **Optimized:** Runs fast on consumer hardware or at scale.
-- **Open-source:** No strings attached.
+|  |  |
```
Comment (carmocca) [inline README.md:54]: It might look better to center this table like the title is

File: README.md [hunk 3]
```diff
@@ -14,183 +14,315 @@
...
-| Vicuna by LMSYS                                                                    | 7B, 13B, 33B                             | [Li et al. 2023](https://lmsys.org/blog/2023-03-30-vicuna/)                                                                  |
+&nbsp;
+## LitGPT supports
 
+✅ &nbsp;[The latest model weights](tutorials/download_model_weights.md): Gemma, Mistral, Mixtral, Phi 2, Llama 2, Falcon, CodeLlama, and [many more](tutorials/download_model_weights.md).
 
-This implementation extends on [Lit-LLaMA](https://github.com/lightning-AI/lit-llama) and [nanoGPT](https://github.com/karpathy/nanoGPT), and it's **powered by [Lightning Fabric](https://lightning.ai/docs/fabric/stable/) ⚡**.
+✅ &nbsp;Optimized and efficient code: Flash Attention v1 and Flash Attention v2, multi-GPU support via fully-sharded data parallelism, [optional CPU offloading](tutorials/oom.md#do-sharding-across-multiple-gpus), and [TPU and XLA support](./xla).
```
Comment (carmocca) [inline README.md:28]: Why mention v1?

```suggestion
✅ &nbsp;Optimized and efficient code: Flash Attention v2, multi-GPU support via fully-sharded data parallelism, [optional CPU offloading](tutorials/oom.md#do-sharding-across-multiple-gpus), and [TPU and XLA support](./xla).
```

File: README.md [hunk 4]
```diff
@@ -14,183 +14,315 @@
...
 
-Install with all dependencies (including CLI, quantization, tokenizers for all models, etc.):
+
+## Installing LitGPT
+
+You can install LitGPT with all dependencies (including CLI, quantization, tokenizers for all models, etc.) using the following pip command:
 
 ```bash
-pip install 'litgpt[all]'
+pip install "litgpt[all]"
```
Comment (carmocca) [inline README.md:75]: zsh friendly

```suggestion
pip install 'litgpt[all]'
```

File: README.md [hunk 5]
```diff
@@ -14,183 +14,315 @@
...
 ```
 
+Alternatively, can install litgpt from a cloned GitHub repository:
+
+```bash
+git clone https://github.com/Lightning-AI/litgpt/blob/wip/config_hub/finetune/llama-2-7b/full.yaml
```
Comment (carmocca) [inline README.md:81]:
```suggestion
git clone https://github.com/Lightning-AI/litgpt
```

File: README.md [hunk 6]
```diff
@@ -14,183 +14,315 @@
...
+cd litgpt
+pip install ".[all]"
```
Comment (carmocca) [inline README.md:83]: Lets suggest editable locally

```suggestion
pip install -e '.[all]'
```

File: README.md [hunk 7]
```diff
@@ -14,183 +14,315 @@
...
 
-To generate text predictions, you need to download the model weights. **If you don't have them, check out our [guide](tutorials/download_model_weights.md).**
+Below is a minimal example to get started with the LitGPT command line interface (CLI), illustrating how to download a model, optionally finetune it using low-rank adaptation (LoRA), and start chatting with it:
 
-Run inference:
 
 ```bash
-litgpt generate base --prompt "Hello, my name is"
+# 1) Download a pretrained model
+python litgpt download --repo_id mistralai/Mistral-7B-v0.1
```
Comment (carmocca) [inline README.md:96]:
```suggestion
litgpt download --repo_id mistralai/Mistral-7B-v0.1
```

File: README.md [hunk 8]
```diff
@@ -14,183 +14,315 @@
...
+
+# 2) Optionally finetune the model
+python litgpt finetune lora \
```
Comment (carmocca) [inline README.md:99]:
```suggestion
litgpt finetune lora \
```

File: README.md [hunk 9]
```diff
@@ -14,183 +14,315 @@
...
+  --checkpoint_dir checkpoints/mistralai/Mistral-7B-v0.1 \
+  --train.micro_batch_size 2 \
+  --lora_r 4 \
+  --precision bf16-true \
+  --data Alpaca2k \
+  --out_dir out/my-finetuned-model
+
+# 3) Chat with the model
+python finetune chat \
```
Comment (carmocca) [inline README.md:108]:
```suggestion
litgpt chat \
```

File: README.md [hunk 10]
```diff
@@ -14,183 +14,315 @@
...
+&nbsp;
+
+## Configuration files for enhanced performance
+
+LitGPT also allows users to use configuration files in YAML format instead of specifying settings via the command line interface and comes with a set of model-specific defaults for good out-of-the-box performance:
+
+
+```bash
+python litgpt finetune lora \
+  --config https://github.com/Lightning-AI/litgpt/blob/wip/config_hub/finetune/llama-2-7b/lora.yaml
```
Comment (carmocca) [inline README.md:127]: I think this only works with raw files. Also, let's link to main

```suggestion
  --config https://raw.githubusercontent.com/Lightning-AI/litgpt/main/config_hub/finetune/llama-2-7b/lora.yaml
```

File: README.md [hunk 11]
```diff
@@ -14,183 +14,315 @@
...
-You can also chat with the model interactively:
+&nbsp;
+
+## Configuration files for enhanced performance
+
+LitGPT also allows users to use configuration files in YAML format instead of specifying settings via the command line interface and comes with a set of model-specific defaults for good out-of-the-box performance:
+
+
+```bash
+python litgpt finetune lora \
```
Comment (carmocca) [inline README.md:126]:
```suggestion
litgpt finetune lora \
```

File: README.md [hunk 12]
```diff
@@ -14,183 +14,315 @@
...
+  --config https://github.com/Lightning-AI/litgpt/blob/wip/config_hub/finetune/llama-2-7b/lora.yaml
+```
+
+For added convenience, you can also manually override config file setting via the CLI:
+
 
 ```bash
-litgpt chat
+python litgpt finetune lora
```
Comment (carmocca) [inline README.md:134]:
```suggestion
litgpt finetune lora 
```

File: README.md [hunk 13]
```diff
@@ -14,183 +14,315 @@
...
+  --config https://github.com/Lightning-AI/litgpt/blob/wip/config_hub/finetune/llama-2-7b/lora.yaml
+```
+
+For added convenience, you can also manually override config file setting via the CLI:
+
 
 ```bash
-litgpt chat
+python litgpt finetune lora 
+  --config https://github.com/Lightning-AI/litgpt/blob/wip/config_hub/finetune/llama-2-7b/lora.yaml \
```
Comment (carmocca) [inline README.md:135]:
```suggestion
  --config https://raw.githubusercontent.com/Lightning-AI/litgpt/main/config_hub/finetune/llama-2-7b/lora.yaml \
```

File: README.md [hunk 14]
```diff
@@ -14,183 +14,315 @@
...
+  --lora_r 4
 ```
 
+You can browse the available configuration files [here](https://github.com/Lightning-AI/litgpt/tree/main/config_hub).
+
 &nbsp;
 
-### Run large models on smaller consumer devices
+> [!TIP] 
+> **Run large models on smaller consumer devices**
```
Comment (carmocca) [inline README.md:144]: The newline doesnt render

```suggestion
> **Run large models on smaller consumer devices:**
```

File: README.md [hunk 15]
```diff
@@ -14,183 +14,315 @@
...
+&nbsp;
+<br>
 &nbsp;
 
-## Finetune the model
+## Customization
 
-We provide a simple training scripts (`litgpt/finetune/*.py`) that instruction-tunes a pretrained model on the [Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset.
-For example, you can either use
+LitGPT supports rich and customizeable [config files](config_hub) to tailor the LLM training to your dataset and hardware needs. Shown below is a configuration file for LoRA finetuning:
```
Comment (carmocca) [inline README.md:154]:
```suggestion
LitGPT supports rich and customizable [config files](config_hub) to tailor the LLM training to your dataset and hardware needs. Shown below is a configuration file for LoRA finetuning:
```

File: README.md [hunk 16]
```diff
@@ -14,183 +14,315 @@
...
 
+> [!TIP] 
+> Unsure about contributing? Check out our [How to Contribute to LitGPT](https://lightning.ai/pages/community/tutorial/how-to-contribute-to-litgpt/) guide.
+
+If you have general questions about building with LitGPT, please [join our Discord](https://discord.gg/VptPCZkGNa).
 
 
 &nbsp;
 
+## Tutorials and how-to guides
```
Comment (carmocca) [inline README.md:315]: Can we make the `tutorials/` directly more prominent? Also include the word "docs" for those who cmd+F it

File: tutorials/models.md
Comment (rasbt) [reply tutorials/models.md:1]: This was accidentally included in the commit

File: tutorials/finetune.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+or Adapter v2 ([Gao et al. 2023](https://arxiv.org/abs/2304.15010)):
+
+```bash
+litgpt finetune adapter_v2
+```
+
+or LoRA ([Hu et al. 2021](https://arxiv.org/abs/2106.09685)):
+
+```bash
+litgpt finetune lora
```
Comment (awaelchli) [inline tutorials/finetune.md:21]: Should maybe lora move up? Right now it looks like it is the least important.
Full finetuning is not mentioned

File: README.md [hunk 18]
```diff
@@ -14,183 +14,315 @@
...
+litgpt finetune lora \
+  --checkpoint_dir checkpoints/mistralai/Mistral-7B-v0.1 \
+  --train.micro_batch_size 2 \
+  --lora_r 4 \
+  --precision bf16-true \
+  --data Alpaca2k \
+  --out_dir out/my-finetuned-model
+
+# 3) Chat with the model
+litgpt finetune chat \
```
Comment (awaelchli) [inline README.md:108]:
```suggestion
litgpt chat \
```

File: README.md [hunk 19]
```diff
@@ -14,183 +14,315 @@
...
+  --checkpoint_dir checkpoints/mistralai/Mistral-7B-v0.1 \
```
Comment (awaelchli) [inline README.md:109]: This checkpoint dir will depend on whether the user chose to run the optional finetuning step above. 
Also, Mistral-7B-v0.1 is not a chat model so I don't expect it to behave well (or does it?). 

For these commands, I think we should decide if an end-to-end example is more important, or whether individual steps like above are better.

**Review by awaelchli — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (rasbt) [reply README.md:54]: Fixed

File: README.md (hunk 5 above)
Comment (rasbt) [reply README.md:81]: Oh ... not sure what happened there.

File: README.md (hunk 19 above)
Comment (rasbt) [reply README.md:109]: Good point. Let's just use chat here and show finetuning further below.

**Comment by rasbt** [comment]
I believe all your points are addressed now.  Many thanks for the feedback! 

Regarding the LoRA example, @awaelchli , this would be the intended use after your auto-merging PR, correct?

```bash
# 1) Download a pretrained model
litgpt download --repo_id microsoft/phi-2

# 2) Finetune the model
litgpt finetune lora \
  --data Alpaca2k \
  --out_dir out/phi-2-lora \
  --checkpoint_dir checkpoints/microsoft/phi-2

# 3) Chat with the model
litgpt chat \
  --checkpoint_dir out/phi-2-lora/final
```

**Comment by awaelchli** [comment]
@rasbt Yes that would work 🎉

**Pull request merged**